DSEGOG-422 save plots on session save

joshdimanteto opened the pull request:
## Description
see title

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes DSEGOG-422

louise-davies committed:
DSEGOG-422 add function to plot window to enable save on session save

joshdimanteto committed:
add e2e tests and unit test

louise-davies commented on e2e/mocked/sessions.spec.ts:
```diff
@@ -0,0 +1,177 @@
+      screenY = 200;
+      screenX = 200;
+      colour = '#41817f';
+  }
```
I wonder if instead of all this complexity of different browsers having different colours - to try and instead do a partial match (e.g. using `toMatchObject` instead of `toEqual` which allows for matching a subset of properties) so we only check for session properties we care about validating or using `expect.objectContaining()` and using e.g. `expect.any(string)` on values that we don't care about.

Also, why are we changing the plot colour in this test anyway? The test is the same if we just select the channel without selecting the colour or changing something easier e.g. the title of the plot.

louise-davies commented on e2e/mocked/sessions.spec.ts:
```diff
@@ -0,0 +1,177 @@
+          axesLabelsVisible: true,
+          gridVisible: true,
+          id: plotId,
+          innerHeight: 400,
+          innerWidth: 600,
```
I think I remember when looking at this when I implemented the saving positions code that there's no way to resize the popup in playwright? I'm not sure if this is still true, if we can resize the window it would be useful to test that here as well

joshdimanteto committed:
Improve e2e tests

- Add tewst to resize the plot popup window
- use objectContaining to check the plot properties in saved session

louise-davies commented:
Add extra e2e test for non-plot windows saving their position/size

joshdimanteto committed:
Add extra test for window position for sessions

joshdimanteto committed:
Merge branch 'develop' into DSEGOG-422-save-plots-on-session-save

louise-davies reviewed (approved):
LGTM
